test/Microsoft.TemplateEngine.Cli.UnitTests/CliMocks/MockNewCommandInput.cs

```diff
@@ -5,86 +5,173 @@
+            {
+                if (_commandOptions.ContainsKey("--author"))
+                {
+                    return _commandOptions["--author"];
+                }
+                return string.Empty;
+            }
+        }
+
+        public string BaselineName
+        {
+            get
+            {
+                if (_commandOptions.ContainsKey("--baseline"))
+                {
+                    return _commandOptions["--baseline"];
+                }
+                return string.Empty;
+            }
+        }
+
+        public bool CheckForUpdates { get; }
+
+        public bool CheckForUpdatesNoPrompt { get; }
+
+        public IReadOnlyCollection<string> Columns
+        {
+            get
+            {
+                if (_commandOptions.ContainsKey("--columns"))
+                {
+                    return _commandOptions["--columns"].Split(",");
+                }
+                return new List<string>();
+            }
+        }
 
         public string ColumnsParseError => throw new NotImplementedException();
 
         public string CommandName => "MockNew";
 
-        public bool ExpandedExtraArgsFiles { get; set; }
+        public bool ExpandedExtraArgsFiles { get; }
 
-        public IList<string> ExtraArgsFileNames { get; set; }
+        public IList<string> ExtraArgsFileNames { get; }
 
         public bool HasColumnsParseError => throw new NotImplementedException();
 
-        public bool HasParseError { get; set; }
+        public bool HasParseError { get; }
 
-        public string HelpText { get; set; }
+        public string HelpText { get; }
 
         // When using this mock, set the inputs using constructor input.
         // This property gets assigned based on the constructor input and the template being worked with.
         public IReadOnlyDictionary<string, string> InputTemplateParams { get; private set; }
 
-        public IList<string> InstallNuGetSourceList { get; set; }
+        public IList<string> InstallNuGetSourceList { get; }
 
-        public bool IsDryRun { get; set; }
+        public bool IsDryRun { get; }
 
-        public bool IsForceFlagSpecified { get; set; }
+        public bool IsForceFlagSpecified { get; }
 
-        public bool IsHelpFlagSpecified { get; set; }
+        public bool IsHelpFlagSpecified => _commandOptions.ContainsKey("--help") || _commandOptions.ContainsKey("-h");
 
-        public bool IsInteractiveFlagSpecified { get; set; }
+        public bool IsInteractiveFlagSpecified { get; }
 
-        public bool IsListFlagSpecified { get; set; }
+        public bool IsListFlagSpecified => _commandOptions.ContainsKey("--list") || _commandOptions.ContainsKey("-l");
 
-        public bool IsQuietFlagSpecified { get; set; }
+        public bool IsQuietFlagSpecified { get; }
 
-        public bool IsShowAllFlagSpecified { get; set; }
+        public bool IsShowAllFlagSpecified { get; }
 
-        public string Language { get; set; }
+        public string Language 
```

Comment:
Most properties in this Type boast the same pattern & could all be simplified via ternary operator
`public string Language => _commandOptions.ContainsKey("--language") ? _commandOptions["--language"] : string.Empty;`